next/components/Molecules/GalleryBanner.tsx

```diff
@@ -1,114 +1,14 @@
-                  <div className="flex h-full items-center justify-center border-2 border-border-dark text-center">
-                    <div>
-                      <p>{t('noImage')}</p>
-                    </div>
-                  </div>
-                </div>
-              )}
-            </div>
-            <div className="w-1/2 lg:h-1/2">
-              {gallery[2]?.Photo ? (
-                <img
-                  onClick={() => setShowModal(true)}
-                  className="h-full w-full cursor-pointer object-cover py-2 lg:pt-0 lg:pl-2"
-                  src={gallery[2].Photo?.data?.attributes?.url}
-                  alt={gallery[2].Photo?.data?.attributes?.alternativeText || 'Gallery'}
-                  aria-hidden="true"
-                />
-              ) : (
-                <div className="hidden h-full py-2 lg:block lg:pt-0 lg:pl-2">
-                  <div className="flex h-full items-center justify-center border-2 border-border-dark text-center">
-                    <div>
-                      <p>{t('noImage')}</p>
-                    </div>
-                  </div>
-                </div>
-              )}
-            </div>
-            <div className="w-1/2 lg:h-1/2">
-              {gallery[3]?.Photo ? (
-                <img
-                  onClick={() => setShowModal(true)}
-                  className="h-full w-full cursor-pointer object-cover pr-2 lg:pr-0 lg:pl-2"
-                  src={gallery[3].Photo?.data?.attributes?.url}
-                  alt={gallery[3].Photo?.data?.attributes?.alternativeText || 'Gallery'}
-                  aria-hidden="true"
-                />
-              ) : (
-                <div className="hidden h-full pr-2 lg:block lg:pr-0 lg:pl-2">
-                  <div className="flex h-full items-center justify-center border-2 border-border-dark text-center">
-                    <div>
-                      <p>{t('noImage')}</p>
-                    </div>
-                  </div>
-                </div>
-              )}
-            </div>
-            {gallery.length > 4 ? (
-              <div className="w-1/2 lg:pl-2">
-                <div
-                  onClick={() => setShowModal(true)}
-                  className="flex h-full cursor-pointer items-center justify-center border-2 border-border-dark text-center"
-                  aria-hidden="true"
-                >
-                  <div>{t('galleryMorePhotos')}</div>
-                </div>
-              </div>
-            ) : (
-              <div className="hidden w-1/2 lg:block lg:pl-2">
-                <div className="flex h-full items-center justify-center border-2 border-border-dark text-center">
-                  <div>{t('noImage')}</div>
-                </div>
-              </div>
-            )}
-          </div>
-        </div>
-      )}
-    </>
-  )
+  return <>{gallery && <ImageGallery images={gallery} variant="below" />}</>
```

Comment:
This will always return `ImageGallery` because empty array is "truthy". You need to check the array length :)

Reply:
Just for code style an readability, we prefer early returns in this case rather than unnecessary fragments. So I would write

```
if (!gallery.length) {
  return null
}

return (
  <ImageGallery ... />
)
```

Reply:
But, if `ImageGallery` can handle empty array, there is no need to check it twice.
So with this thought, is there a need to have `GalleryBanner`?
I see two ways:
- keep `GalleryBanner` with some purpose - change props back to accept "section props" and parse images in `GalleryBanner`, not in `Sections`
- remove `GalleryBanner` completely and use `ImageGallery` directly in `Sections`.

(thinking about the name...if it represents section, `GallerySection` is suitable name :D)